core/trino-main/src/main/java/io/trino/operator/TopNRankingOperator.java

```diff
@@ -272,6 +338,11 @@ public void addInput(Page page)
         checkState(unfinishedWork == null, "Cannot add input with the operator when unfinished work is not empty");
         checkState(outputIterator == null, "Cannot add input with the operator when flushing");
         requireNonNull(page, "page is null");
+        checkState(groupedTopNBuilder == null || !groupedTopNBuilder.isFull(), "TopN buffer is full");
```

Comment:
If you add the isBuilderFull() method I suggested above, this would just be:

```
checkState(!isBuilderFull(), "TopN buffer is already full")
```

Reply:
Done